bootstrap-app/src/cmr/bootstrap/services/dispatch/core_async_dispatch.clj

```diff
@@ -0,0 +1,143 @@
+(ns cmr.bootstrap.services.dispatch.core-async-dispatch
```

Comment:
Some notes on this -- and, again, I think you may be thinking of these things for a future PR:

* the filename is a little redundant, given the directory name -- I'd rename it to `.../dispatch/async/clj`
* a common idiom in Clojure is for the protocol API to be defined in a `core` ns, e.g., `.../dispatch/core.clj` which is used by the rest of the application and then for implementations of this to go in `.../dispatch/impl`, 
* as such I would really rename this ns to `.../dispatch/impl/async.clj`

Note that the *only* place that would ever get `require`d would be `dispatch.core` (with the possible exception of unit tests). See my comments below on other things that need to be moved that will help this make more sense (e.g., the `extend` calls).